docs/dev_guide/2.quick_start.md

```diff
@@ -0,0 +1,160 @@
+
+## Asking Questions
+
+Before asking a question, make sure you have:
+
+- Searched open and closed:
+  - [KCLVM GitHub Issues](https://github.com/KusionStack/KCLVM/issues?utf8=%E2%9C%93&q=is%3Aissue)
+
+- Read the documentations:
+  - [KCLVM Documents](https://kusionstack.io/docs/reference/lang/lang/tour)
+  - [KCLVM Readme](https://github.com/KusionStack/KCLVM)
+
+If you have any questions about `KCLVM`, you are welcome to ask your questions in [KCLVM Github Issues](https://github.com/KusionStack/KCLVM/issues). When you ask a question, please describe the details as clearly as possible so that others in the KCLVM community can understand, and you *MUST* be polite and avoid personal attack and avoid not objective comparison with other projects.
+
+## Cloning and Building `KCLVM`
+
+### Dependencies
+
+#### macOS and OS X
+
+- `git`
+- `Python 3.7+`
+- `Rust 1.60+`
+- `openssl@1.1`
+- `LLVM 12`
+
+You can install `openssl@1.1` using `brew`:
+
+```sh
+brew install openssl@1.1
+```
+
+Besides, you'll need LLVM installed and `llvm-config` in your `PATH`. Just download from [LLVM 12](https://releases.llvm.org/download.html) or install `llvm@12` using `brew`.
+
+```sh
+brew install llvm@12
+```
+
+Add the LLVM installation location to `LLVM_SYS_120_PREFIX` and the `$PATH`.
+
+```
+export LLVM_SYS_120_PREFIX=<your LLVM 12 install location>
+export PATH=<your LLVM 12 install location>/bin:$PATH
+```
+
+#### Linux
+
+- `git`
+- `Rust 1.60+`
+- `Python3 Building Dependencies`
+- `LLVM 12`
+
+For UNIX based systems, you can run:
+
+```
+yum groupinstall -y "Development Tools"
+yum install -y gcc patch libffi-devel python-devel zlib-devel bzip2-devel ncurses-devel sqlite-devel 
+yum install -y libpcap-devel xz-devel readline-devel tk-devel gdbm-devel db4-deve
+yum -y install yum-utils
+yum-builddep -y python3
+yum install -y zlib* 
+yum install -y openssl-devel
+yum install -y glibc-static
+
+# clang-12 & llvm-12
+RUN yum -y install clang
+RUN clang --version
+RUN yum -y install llvm-devel
+RUN yum -y install libffi-devel
+RUN ln -s /usr/lib64/libtinfo.so.6 /usr/lib64/libtinfo.so
+```
+
+On Debian, Ubuntu, and other apt based systems, you can run:
+
+```
+apt-get update
+
+apt-get install -y git wget curl
+apt-get install -y make gcc patch 
+apt-get install -y python-dev libffi-dev
+apt-get install -y zlib1g-dev ncurses-dev build-essential libncurses5-dev libgdbm-dev libnss3-dev libssl-dev libreadline-dev libffi-dev
+
+# clang-12 & llvm-12
+RUN apt-get install -y clang-12 lld-12
+RUN ln -sf /usr/bin/clang-12   /usr/bin/clang
+RUN ln -sf /usr/bin/wasm-ld-12 /usr/bin/wasm-ld
+```
+
+#### Windows
+
+- `git`
+- `Rust 1.60+`
+- `Python 3.7+`
+- `LLVM 12`
+
+Please add the LLVM installation location to `LLVM_SYS_120_PREFIX` and the `$PATH`.
+
+#### Docker
+
+Use the image `kusionstack/kclvm-builder`
```

Comment:
+ Change the desc with a runnable cli
+ Make the docker-env as the recommended method

Reply:
> * Change the desc with a runnable cli
> * Make the docker-env as the recommended method

Modified, PTAL